### PR TITLE
NIFI-8353: When receiving data via load-balanced connection, throw an…

### PR DESCRIPTION
**Pull request by markap14**
… Exception (resulting in an ABORT_TRANSACTION status code) when attempting to add received FlowFiles to the FlowFile queue, if the node is not currently connected to cluster. In this case, ensure that we remove the received FlowFiles from the FlowFile Repository and emit a DROP event to the repository (with appropriate details) to coincide with the already-committed Provenance events. Also ensure that when a node is disconnected that we don't keep sending to that node until it reconnects. While testing the fixes via LoadBalancedQueueIT, also noticed that unit test logging was not working properly in nifi-framework-core because of the (erroneous) dependency on logback-class; removed this dependency and updated unit tests that were created that depended on its existence.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by bbende** [comment]
Reviewing

File: nifi-framework-api/src/main/java/org/apache/nifi/controller/queue/IllegalClusterStateException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.controller.queue;
+
```
Comment (ottobackwards) [inline nifi-framework-api/src/main/java/org/apache/nifi/controller/queue/IllegalClusterStateException.java:19]: Sorry if this is stupid, but this seems more like an IllegalClusterOperation.   The exception is thrown when you attempt an operation, not when you enter an illegal state.

Also, there is no knowing the "state" you are in.

Comment (markap14) [reply nifi-framework-api/src/main/java/org/apache/nifi/controller/queue/IllegalClusterStateException.java:19]: The Exception denotes that the operation cannot performed when the cluster is in the given state. It has the same semantics, in that way, as IllegalStateException - I.e., IllegalStateException does not imply that the object is in an illegal state, but rather that the operation you're trying to perform cannot be performed in the current state. This is analogous. The operation cannot be performed at the current time due to the state of the cluster.

There is no need to know, programmatically what state the cluster is in. The message provided to the Exception can provide information for users to understand the context.

**Comment by bbende** [comment]
+1 Verified disconnecting/off-loading a node in a local cluster does not receive data from a load balanced connection, will merge

**Pull request merged**